### PR TITLE
build(tree-comparison): use node16 module resolution

### PR DESCRIPTION
**Pull request by DLehenbauer**
Semi-automatic PR to switch to node16 module resolution.

Fixed 'ISharedTree' -> 'ITree'.  (ISharedTree is an internal interface, ITree is public).

### CODE REVIEW

File: examples/apps/tree-comparison/src/model/newTreeInventoryList.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ import {
 	SchemaFactory,
 	SharedTree,
 	NodeFromSchema,
-	type ISharedTree,
+	type ITree,
```
Comment (DLehenbauer) [inline examples/apps/tree-comparison/src/model/newTreeInventoryList.ts:12]: ISharedTree is an internal interface that extends ITree.

ITree is public.

File: examples/apps/tree-comparison/src/model/newTreeInventoryList.ts [hunk 2]
```diff
@@ -101,8 +101,11 @@ class NewTreeInventoryItem extends TypedEmitter<IInventoryItemEvents> implements
 }
 
 export class NewTreeInventoryList extends DataObject implements IInventoryList {
-	private _sharedTree: ISharedTree | undefined;
-	private get sharedTree(): ISharedTree {
+	on(event: "itemAdded" | "itemDeleted", listener: (item: IInventoryItem) => void): this {
+		throw new Error("Method not implemented.");
+	}
```
Comment (DLehenbauer) [inline examples/apps/tree-comparison/src/model/newTreeInventoryList.ts:106]: Not sure why we were getting away with this before, but the TSC error seems legit:
```
Class 'NewTreeInventoryList' incorrectly implements interface 'IInventoryList'.
  Property 'on' is missing in type 'NewTreeInventoryList' but required in type 'IInventoryList'.ts(2420)
modelInterfaces.ts(47, 2): 'on' is declared here.
```
@ChumpChief - FYI.

Comment (DLehenbauer) [reply examples/apps/tree-comparison/src/model/newTreeInventoryList.ts:106]: A partial explanation:
* NewTreeInventoryList transitively inherits an implementation of TypedEventEmitter via DataObject.
* However, DataObject has an empty event map by default.
    * Subclasses provide the event map by passing a generic argument (i.e., ... extends DataObject<{ Events: { itemAdded: "itemAdded", itemRemoved: "itemRemoved" }}>
    * Without an event map, the signature is '.on(eventKey: never, ...)'

...so TypeScript was previously only missing the fact that the inheritted '.on(never)' does not satisfy '.on(itemAdded | itemRemoved)'.

**Review by tylerbutler — APPROVED** [review]
🚀

**Pull request merged**